include/openssl/core_names.h

```diff
@@ -34,6 +34,31 @@ extern "C" {
  */
 #define OSSL_PROV_PARAM_BUILDINFO   "buildinfo"
 
+/*
+ * The module filename (Used for FIPS Self Testing)
+ * Type: OSSL_PARAM_UTF8_STRING
+ */
+#define OSSL_PROV_PARAM_MODULE_FILENAME "module-filename"
+/*
+ * The calculated MAC of the module file (Used for FIPS Self Testing)
+ * Type: OSSL_PARAM_UTF8_STRING
+ */
+#define OSSL_PROV_PARAM_MODULE_MAC      "module-checksum"
+/*
+ * A version number for the fips install process (Used for FIPS Self Testing)
+ * Type: OSSL_PARAM_UTF8_STRING
+ */
+#define OSSL_PROV_PARAM_INSTALL_VERSION "install-version"
+/*
+ * The calculated MAC of the install status indicator (Used for FIPS Self Testing)
+ * Type: OSSL_PARAM_UTF8_STRING
+ */
+#define OSSL_PROV_PARAM_INSTALL_MAC     "install-checksum"
+/*
+ * The install status indicator (Used for FIPS Self Testing)
+ * Type: OSSL_PARAM_UTF8_STRING
+ */
+#define OSSL_PROV_PARAM_INSTALL_STATUS  "install-status"
```

Comment:
Do these defines belong in core-names.h? IIUC libcrypto has no knowledge of these parameters, and most modules won't see these values. They're just entries in a config file that get passed to the module, and its only the FIPS module that needs to know about them. So perhaps they belong in a fips module specific header file?

The exception I supposed is `OSSL_PROV_PARAM_MODULE_FILENAME` which libcrypto does know about, and it gets sent to all modules (including non-fips ones).

Reply:
Moved into include/openssl/fips.h
An openssl install app can use them (which is why I put them at this level)

Reply:
fips_names.h, perhaps?

Reply:
Changed name. (I was thinking I could shove other stuff in there later)